### ISSUE TITLE
Geo Replication Topic Policy

### ISSUE THREAD

**Issue by kuskmen** [apache/pulsar#11608]
**Is your feature request related to a problem? Please describe.**
This feature request is not related to any problem, at least not that I am familiar with.

**Describe the solution you'd like**
I think the current architecture and way of configuring pulsar it will be nice to introduce geo-replication policy on topic level. Unfortunately, I dont know the details of the internal implementation of pulsar and if you have considered such feature before and if you did what would be the impact of it.

**Describe alternatives you've considered**
There is a way of making the so-called _[selective replication](https://pulsar.apache.org/docs/en/administration-geo/#selective-replication)_ which ideally mimics the functionality that one geo-replication topic level policy would, but I think its rather odd making publishers in the clients to configure which clusters they should use. This should really be ops part since this is something you shouldn't (in fact maybe never) really change that often _after_ you've configured it once in your environment.

Ofc there is another alternative - creating a dedicated namespace for this topic but this will only prove the fact that such configuration is needed on topic level. (imho)

Bottom line, I like where Pulsar is heading in terms of configuration and I think it should make sense to have one of its key functionalities able to be configured as flexible as possible, also what I said, I really think geo-replication should remain in the broker/server/ops domain of pulsar and developers (like me) shouldn't worry about this in the code. Not to mention that having to know which clusters you are connecting to seems like an abstraction leak to me.

**Comment by codelipenghui** [comment]
The issue had no activity for 30 days, mark with Stale label.

**Comment by tisonkun** [comment]
Closed as stale and no one works on it. Please create a new issue if it's still relevant to the maintained versions.
